### PR TITLE
[Java] Add Java release guideline.

### PR DESCRIPTION
**Pull request by jovany-wang**
## Why are these changes needed?
Add Java release guideline to help us release Ray Java process.

## Related issue number
#18486

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: java/java-release-guide.md [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+```bash
+export OSSRH_KEY=xxx
+export OSSRH_TOKEN=xxx
+export TRAVIS_BRANCH=releases/x.y.z  # Your releasing version number
+export TRAVIS_COMMIT=xxxxxxxxxxx     # The commit id
+git checkout $TRAVIS_COMMIT
+sh java/build-jar-multiplatform.sh multiplatform
+export GPG_SKIP=false
+cd java && mvn versions:set -DnewVersion=x.y.z && cd -
+cd streaming/java && mvn versions:set -DnewVersion=x.y.z && cd -
```
Comment (carsonwang) [inline java/java-release-guide.md:29]: Streaming has been removed so it seems this is not needed any more?

File: java/java-release-guide.md [hunk 2]
```diff
@@ -0,0 +1,51 @@
+# Ray Java Release Guideline
+
+This document help Ray Java developers release Ray Java.  
+Note that we assume you release it on your local laptop.
+
+## 1. Install GPG
+**If you have isntalled GPG, you can skip this step.**
```
Comment (carsonwang) [inline java/java-release-guide.md:7]: isntalled  -> installed

**Review by carsonwang — COMMENTED** [review]
Thanks for improving the Java release. Right now, there is still gap between the Python release and Java release. For example ray-1.10.0 Java release is still not available on Maven though the python package has been released for a few days. Are they released by different person? Is there a plan to improve this so they can be released at the same time?

**Comment by jovany-wang** [comment]
@carsonwang Before 1.11.0, Java part is released by myself for some reasons. From 1.11.0, Java part will be released in the whole releasing process. 

BTW, Ray Java 1.10.0 is going to be released by the end of this week.

**Comment by jovany-wang** [comment]
@carsonwang @chaokunyang Could you take one more look?

**Comment by jovany-wang** [comment]
@carsonwang FYI: Ray Java 1.10.0 was released.

**Review by chaokunyang — APPROVED** [review]
LGTM

**Comment by carsonwang** [comment]
Thanks! LGTM

**Comment by jovany-wang** [comment]
CI failure is unrelated because this is only adding a `.md` file.

**Pull request merged**